### PR TITLE
Added Unsupported features, Unsupported datatypes and Schema objects summary in the assessment report using analyze schema code path

### PR DESCRIPTION
**Pull request by sanyamsinghal**
Sample HTML report:
image

### CODE REVIEW

**Comment by sanyamsinghal** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/

**Comment by sanyamsinghal** [comment]
https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/2284/

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 1]
```diff
@@ -182,7 +193,7 @@ func gatherAssessmentDataFromPG() (err error) {
 	}
 
 	homebrewVoyagerDir := fmt.Sprintf("yb-voyager@%s", utils.YB_VOYAGER_VERSION)
-	gatherAssessmentDataScriptPath := filepath.Join("/", "etc", "yb-voyager", "gather-assessment-data", "postgresql", "yb-voyager-gather-assessment-data.sh")
+	gatherAssessmentDataScriptPath := filepath.Join("/", "etc", "yb-voyager", "gather-assessment-data", "postgresql", "yb-voyager-pg-gather-assessment-data.sh")
```
Comment (amit-yb) [inline yb-voyager/cmd/assessMigrationCommand.go:196]: When you know all the components of a path, there is no need of `filepath.Join()`. You can simply write it as a constant string:
```
gatherAssessmentDataScriptPath := "/etc/yb-voyager/gather-assessment-data/postgresql/yb-voyager-pg-gather-assessment-data.sh"
```

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 2]
```diff
@@ -214,55 +225,222 @@ func gatherAssessmentDataFromPG() (err error) {
...
+	unsupportedFeatures, err := fetchUnsupportedFeaturesForPG(analyzeSchemaReport)
+	if err != nil {
+		return fmt.Errorf("failed to fetch unsupported features: %w", err)
+	}
+	assessmentReport.UnsupportedFeatures = unsupportedFeatures
+
+	return nil
+}
+
+func fetchUnsupportedFeaturesForPG(analyzeSchemaReport utils.SchemaReport) ([]string, error) {
```
Comment (amit-yb) [inline yb-voyager/cmd/assessMigrationCommand.go:321]: How about something much simpler:
```
type UnsupportedFeature struct {
	FeatureName string
	ObjectNames []string
}

func fetchUnsupportedFeaturesForPG(analyzeSchemaReport utils.SchemaReport) ([]UnsupportedFeature, error) {
	log.Infof("fetching unsupported features for PG...")
	result := make([]UnsupportedFeature, 0)
	filterIssues := func(featureName, issueReason string) {
		log.Info("filtering issues for feature: ", featureName)
		m := make(map[string][]string)
		for _, issue := range analyzeSchemaReport.Issues {
			if strings.Contains(issue.Reason, issueReason) {
				m[featureName] = append(m[featureName], issue.ObjectName)
			}
		}
		for featureName, objectNames := range m {
			result = append(result, UnsupportedFeature{featureName, objectNames})
		}
	}
	filterIssues("GIN indexes", GIST_INDEX_ISSUE_REASON)
	filterIssues("Constraint triggers", CONSTRAINT_TRIGGER_ISSUE_REASON)
	filterIssues("Inheritance", INHERITANCE_ISSUE_REASON)
	filterIssues("Stored generated columns", STORED_GENERATED_COLUMN_ISSUE_REASON)
	return result, nil
}
```

**Review by amit-yb — COMMENTED** [review]
Reviewed the assessMigrationCommand.go

**Review by amit-yb — APPROVED** [review]
LGTM with minor comments.

File: yb-voyager/cmd/assessMigrationCommand.go [hunk 3]
```diff
@@ -43,11 +43,16 @@ type AssessmentReport struct {
...
 
-	UnsupportedFeatures []string `json:"UnsupportedFeatures"`
+	UnsupportedFeatures []UnsupportedFeature `json:"UnsupportedFeatures"`
 
 	Recommendations migassessment.AssessmentRecommendations `json:"Recommendations"`
 }
 
+type UnsupportedFeature struct {
+	FeatureName string   `json:"feature_name"`
+	ObjectNames []string `json:"object_names"`
```
[the remark this answers is not in the document]
Comment (sanyamsinghal) [reply yb-voyager/cmd/assessMigrationCommand.go:53]: agree. There are mixed casing across analyzeSchema structs also.
Let me do that in a separate PR - will go with CamelCase everywhere.

**Pull request merged**